ingest/rules/nextclade.smk

```diff
@@ -98,6 +98,69 @@ rule append_nextclade_columns:
         > {output.metadata_all}
         """
 
+rule calculate_gene_coverage:
+    """
+    Calculate the coverage of the gene of interest
+    """
+    input:
+        nextclade_translation="data/translations/seqs_{serotype}.gene.{gene}.fasta",
+    output:
+        gene_coverage="data/translations/gene_coverage_{serotype}_{gene}.tsv",
+    wildcard_constraints:
+        serotype=SEROTYPE_CONSTRAINTS,
+    shell:
+        """
+        python bin/calculate-gene-converage-from-nextclade-translation.py \
+          --fasta {input.nextclade_translation} \
+          --out-col {wildcards.gene}_coverage \
+          > {output.gene_coverage}
+        """
+
+rule aggregate_gene_coverage_by_gene:
+    """
+    Aggregate the gene coverage results by gene
+    """
+    input:
+        gene_coverage=expand("data/translations/gene_coverage_{serotype}_{{gene}}.tsv", serotype=SUPPORTED_NEXTCLADE_SEROTYPES),
+    output:
+        gene_coverage_all="results/gene_coverage_all_{gene}.tsv",
+    shell:
+        """
+        # Capture header for tsv
+        head -n1 {input.gene_coverage[0]} > {output.gene_coverage_all}
+        # Capture rest of tsv lines from all files
+        for FILE in {input.gene_coverage}; do
+            awk 'NR>1' $FILE >> {output.gene_coverage_all}
+        done
+        """
+
+rule append_gene_coverage_columns:
+    """
+    Append the gene coverage results to the metadata
+    """
+    input:
+        metadata="data/metadata_nextclade.tsv",
+        gene_coverage=expand("results/gene_coverage_all_{gene}.tsv", gene=config["nextclade"]["gene"])
+    output:
+        metadata_all="results/metadata_all.tsv",
+    params:
+        id_field=config["curate"]["id_field"],
+    shell:
+        """
+        cp {input.metadata} {output.metadata_all}
+        for FILE in {input.gene_coverage}; do
+            export append_col=`awk 'NR==1 {{print $2}}' $FILE`
```

Comment:
I'm not a huge fan of selecting columns by their order number since this can potentially result in silent errors if someone changes the column order upstream and didn't realize it needed to be updated here.

Maybe this could use a wildcard in the field selection:
```
--append-fields '*_coverage'
```

Reply:
Haha, I'm not a fan of alphabetical or random order from a [Human-Computer-Interaction (HCI)](https://www.interaction-design.org/literature/topics/human-computer-interaction) design perspective especially when we have easy-to-use tools like ["tsv-select" to reorder columns](https://github.com/nextstrain/zika/blob/5d9a9670ca59bf3cd65376d9fe3f9f791bd77543/phylogenetic/rules/merge_sequences_usvi.smk#L52) and [configs to set the the column order](https://github.com/nextstrain/zika/blob/5d9a9670ca59bf3cd65376d9fe3f9f791bd77543/ingest/defaults/config.yaml#L96-L112) across all rules.

From an HCI perspective, a way we can encode the importance of a metadata column is via the ordering in which it is presented. For example, from influenza, I would expect the design to be one of the following:

1. The surface proteins listed first (e.g. HA, NA) then the rest of the proteins in decending length order (e.g. PB2, PB1, PA, NP, M, NS)
2. The proteins listed in decending length order (e.g. PB2, PB1, PA, HA, NP, NA, M, NS). For example, see [Table 1](https://journals.plos.org/plosone/article/figure?id=10.1371/journal.pone.0020823.t001) in [Fulvini et al, 2011](https://pubmed.ncbi.nlm.nih.gov/21695145/).
3. Another order in which the biological question of interest is emphasized.

Applying this to Dengue, we expect the surface protein "E" to be potentially more antigenically informative than the non structural proteins (e.g. "NS1"). This is alphabetical but only by accident. Since Dengue is single stranded (not segmented) another option is in the order of the genes on the strand. See [Figure 1 in Nanaware et al, 2021](https://www.mdpi.com/1999-4915/13/10/1967) for such an ordering.

I agree that an alphabetical list is a reasonable and standard guess of computer scientists when unfamiliar with the field of interest; but, a domain scientist should be able to set a reasonable gene order later on.

Reply:
Ah you were specifically referring to the awk statement `awk 'NR==1 {{print $2}}`. I agree that `--append-fields '*_coverage'` is more generalized and less error prone.

Sorry, I'm going to blame my initial reaction to jetlag and past frustration with reading TSV files with random column order. :P

Fixed in 775ca33e2cfbc5fd247b2c339cb72cc5ed8d553b